tables/tabulate-molecular-alterations.R

```diff
@@ -0,0 +1,264 @@
+
+# The final output should only tabuluate alterations for N=1074 tumors 
+# We'll need to identify those samples, including their sample_id and biospecimen.
+
+# In addition, we'll note if a sample_id is ambiguous. Ambiguous sample_id's will 
+# have more than 2 rows associated with it in the histologies file when looking at 
+# tumor samples -- that means we won't necessarily be able to determine **when an 
+# WGS/WXS assay maps to an RNA-seq assay** for the purpose of tabulating molecular alterations
+
+tumor_sample_ids_df <- histologies_df %>%
+  dplyr::filter(sample_type == "Tumor",
+                composition == "Solid Tissue") %>%
+  # keep these columns of interest moving forward:
+  dplyr::select(sample_id, 
+                Kids_First_Biospecimen_ID, 
+                cancer_group_display, 
+                broad_histology_display, 
+                experimental_strategy, 
+                germline_sex_estimate) %>%
+  dplyr::distinct() %>%
+  # annotate if it is ambiguous or: any rows with >2 sample_ids
+  # Associated with 2 sample_ids that have the same experimental strategy
+  dplyr::add_count(sample_id) %>%
+  dplyr::mutate(ambiguous = n > 2) %>% 
+  # now, annotate if it has multiple rows of SAME experimental_strategy (needed for later wrangling)
+  dplyr::add_count(sample_id, experimental_strategy) %>%
+  dplyr::mutate(multiple_exp_strategy = n >= 2) %>% 
+  # remove temporary counting column
+  dplyr::select(-n) %>%
+  # add a modality column for later bookkeeping
+  dplyr::mutate(modality = dplyr::case_when(
+    experimental_strategy == "RNA-Seq" ~ "RNA",
+    experimental_strategy == "WGS"     ~ "DNA",
+    experimental_strategy == "WXS"     ~ "DNA",
+    experimental_strategy == "Targeted Sequencing" ~ "DNA"
+  )) %>%
+  # remove experimental_strategy to enable later joining
+  dplyr::select(-experimental_strategy)
+
+
+# Check that this is equal to `openpbta_n_tumors`
+if (!(length(unique(tumor_sample_ids_df$sample_id))) == openpbta_n_tumors) {
+  stop("Bad sample_id parsing.")
+} 
+
+# Pull out biospecimen id column for convenience 
+bs_ids <- tumor_sample_ids_df %>%
+  dplyr::pull(Kids_First_Biospecimen_ID) %>%
+  unique() 
+
+# Filter dfs to those ids:
+histologies_df <-  dplyr::filter(histologies_df, Kids_First_Biospecimen_ID %in% bs_ids) 
+maf_df         <-  dplyr::filter(maf_df, Tumor_Sample_Barcode %in% bs_ids) 
+fusion_df      <-  dplyr::filter(fusion_df, Sample %in% bs_ids) 
+cnv_df         <-  dplyr::filter(cnv_df, biospecimen_id %in% bs_ids) 
+
+# What are all the bs ids that end up being considered? 
+# This variable can be used when joining things back up
+final_bs_ids <- length(unique(
+  c(
+    maf_df$Tumor_Sample_Barcode,
+    fusion_df$Sample,
+    cnv_df$biospecimen_id
+)))
+
+
+
+#### Prepare maf, cnv, fusion data ------------------
+
+maf_df <- prepare_maf(maf_df)
+fusion_df <- prepare_fusion(fusion_df)
+cnv_df <- prepare_cnv(cnv_df)
+
```

Comment:
Each of these functions is used only once, so I'm not sure the value of having them separated as functions, especially in a separate file. At the least, I would like to see here something about what this preparation is: what are the changes that are being made by these functions?

Reply:
The value of having them in separate functions is mostly for my brain, less clutter during coding. All of that code was all originally in this script, but as the script got bigger and bigger....

Can definitely de-function them in the end, and I'll add some more comments here.